src/utils/logging.jl

```diff
@@ -1,6 +1,187 @@
 
 import Logging
 
+# These use PascalCase to avoid clashing with source filenames.
+const LOG_GROUP_PARSING = :Parsing
+const LOG_GROUP_RECORDER = :Recorder
+const LOG_GROUP_SERIALIZATION = :Serialization
+const LOG_GROUP_SYSTEM = :System
+const LOG_GROUP_TIME_SERIES = :TimeSeries
+
+# Try to keep this updated so that users can check the known groups in the REPL.
+const LOG_GROUPS = (
+    LOG_GROUP_PARSING,
+    LOG_GROUP_RECORDER,
+    LOG_GROUP_SERIALIZATION,
+    LOG_GROUP_SYSTEM,
+    LOG_GROUP_TIME_SERIES,
+)
+const SIIP_LOGGING_CONFIG_FILENAME = joinpath(
+    abspath(joinpath(dirname(Base.find_package("InfrastructureSystems")))),
+    "utils",
+    "logging_config.toml",
+)
+
+const LOG_LEVELS = Dict(
+    "Debug" => Logging.Debug,
+    "Info" => Logging.Info,
+    "Warn" => Logging.Warn,
+    "Error" => Logging.Error,
+)
+
+"""Contains information describing a log event."""
+mutable struct LogEvent
```

Comment:
I had to do some refactoring to get struct composition to work. Sorry for the churn, but most of this has no logic change.